### PR TITLE
Created Open Shelters List View

### LINKED ISSUE #393: Volunteer Shelter List View: Open Shelters in Descending Date Order
name: Feature request
about: Suggest an idea for this project
title: Volunteer Shelter List View: Open Shelters in Descending Date Order
labels: enhancement, frontend, volunteer-experience
assignees: ''

Is your feature request related to a problem?
Volunteers currently cannot easily view which shelters are open on upcoming dates in a simple, streamlined format. Navigating by specific days or filters can make planning less efficient.

Describe the solution you'd like
Implement a list-based view that displays shelters grouped by date in descending order (most recent/upcoming dates first), without requiring any filters or calendar interaction.

Display dates in descending order (latest to earliest)
Under each date, list all shelters that are open
Include shelter name and basic location details
Present information in a clean, scrollable list format
No filters or date selection required—users can simply browse the list

Describe alternatives you've considered

Calendar-based view with selectable dates
Day-of-week dropdown filter
Static grouped list by weekday

Additional context
This approach simplifies the user experience by allowing volunteers to quickly scan upcoming open shelters without interacting with a calendar or filters, making shift planning faster and more intuitive.

### PR DESCRIPTION
**Pull request by orhankoylu**
Fixes #393

What was changed?

Updated the volunteer dashboard experience for open shelters and related volunteer scheduling flows. The volunteer-facing open-shelters feature was changed from a calendar-based picker into a list-based view grouped by date, and the upcoming-shifts screen was cleaned up to remove the duplicate legacy table. I also adjusted shelter shift creation so typed instructions are preserved when creating shifts.

Why was it changed?

Issue #393 asked for a simple list view where volunteers can browse upcoming open shelters without interacting with a calendar or filters. The previous implementation required clicking a date, which did not match the story. The updated version fixes that by automatically showing upcoming open shelters grouped by date in descending order. While validating the flow, I also fixed two related UX/behavior issues: weekday visibility in the prior open-shelters UI, duplicate upcoming-shifts tables, and instructions being dropped when creating shifts unless a recurring toggle was enabled.

How was it changed?

Modified client_app/src/components/volunteer/OpenSheltersCalendar.js to replace the calendar interaction with a grouped, scrollable list of open shelters by date. Updated client_app/src/utils/openShelterCalendar.js to add grouping and descending-date sorting logic, and extended client_app/src/utils/openShelterCalendar.test.js to cover that behavior. Refreshed client_app/src/styles/volunteer/OpenSheltersCalendar.css so the page matches the new list layout. Updated client_app/src/components/NavigationConfig.js so the volunteer nav label now reflects the list-based feature.

Also cleaned up client_app/src/components/volunteer/Commitments.js by removing the older duplicated upcoming-shifts table and keeping a single list/calendar flow. In client_app/src/components/shelter/ScheduleManager.js, changed shift creation so instructions are saved on created shifts even when the recurring-instructions toggle is off.

Screenshot 2026-04-13 at 11 28 18 AM

### CODE REVIEW

File: client_app/src/components/volunteer/OpenSheltersCalendar.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    const fetchCalendarData = async () => {
+      try {
+        const [shelterData, shiftData] = await Promise.all([
+          shelterAPI.getShelters(),
+          serviceShiftAPI.getFutureShifts(),
+        ]);
+
+        setShelters(shelterData);
+        setFutureShifts(shiftData);
+      } catch (error) {
```
Comment (Supraja050202) [inline client_app/src/components/volunteer/OpenSheltersCalendar.js:25]: Currently, errors are logged but not surfaced to users, which may result in an indefinite loading state. Introduce an error state and render fallback UI.

File: client_app/src/utils/openShelterCalendar.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  ];
+
+  const shifts = [
+    { shelter_id: 's1', shift_start: new Date('2026-04-15T09:00:00').getTime() },
+    { shelter_id: 's1', shift_start: new Date('2026-04-15T14:00:00').getTime() },
+    { shelter_id: 's2', shift_start: new Date('2026-04-15T11:30:00').getTime() },
+    { shelter_id: 's3', shift_start: new Date('2026-04-16T08:00:00').getTime() },
+    { shelter_id: 's3', shift_start: 'invalid' },
+  ];
+
```
Comment (Supraja050202) [inline client_app/src/utils/openShelterCalendar.test.js:21]: Good baseline coverage, but several important edge cases are missing:

Missing Tests
Empty shifts array
Null/undefined shelters
Duplicate shelter IDs per day
Boundary timestamps (e.g., current time)
Large dataset performance

**Review by Supraja050202 — CHANGES_REQUESTED** [review]
Great work on implementing the Open Shelters List View and related functionality. The feature adds clear value, and there are several good practices already in place (responsive design, useMemo optimization, accessibility considerations, and clean utility structure).

That said, I’ve identified a few areas where improvements would significantly enhance maintainability, robustness, and user experience.

**Review by Supraja050202 — APPROVED** [review]
Approved!
